packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -5430,6 +5430,22 @@ class SemanticsConfiguration {
 
   bool _hasFlag(SemanticsFlag flag) => (_flags & flag.index) != 0;
 
+  bool get _hasExplicitRole {
+    if (_role != SemanticsRole.none) {
+      return true;
+    }
+    if (_hasFlag(SemanticsFlag.isTextField) ||
+        _hasFlag(SemanticsFlag.isHeader) ||
+        _hasFlag(SemanticsFlag.isSlider) ||
+        _hasFlag(SemanticsFlag.isLink) ||
+        _hasFlag(SemanticsFlag.scopesRoute) ||
+        _hasFlag(SemanticsFlag.isImage) ||
+        _hasFlag(SemanticsFlag.isKeyboardKey)) {
```

Comment:
Should we also have `isButton` here?

Reply:
no, an image can also be a button. same for keyboard key. button is more of a trait now...

now I think of it, there is also a searchBox/combobox which can be compatible with istextfield... it almost feels like the compatibility is per flag and role based..

Reply:
we may need to make the role and enhance enum to be able to enter compatible flags

Reply:
discussed offline, for role that may be compatibile, we could (a) factor out some aspect of it to a semanticsproperty(e.g searchable) (b) comes up with a way to customize merging for each enum (e.g. an enhance enum that include other mergable roles).

we can deal with them case by case when these kind of problem raise up. For now, we will go ahead with this pr